oauthenticator/github.py

```diff
@@ -195,7 +197,9 @@ async def authenticate(self, handler, data=None):
         # If a public email is not available, an extra API call has to be made
         # to a secondary endpoint using the access token.
         # see https://github.com/jupyterhub/oauthenticator/issues/438
-        if not auth_state['github_user']['email']:
+        if not auth_state['github_user']['email'] and (
+            'user' in scopes or 'user:email' in scopes
+        ):
```

Comment:
`"user" in ["read:user"]` evaluates to False, so this should also check for `read:user`

Reply:
the normalization function would return `user` according to their example i think.

Reply:
`user` is read and write, so if you have requested `user` and `read:user` then `read:user` is normalized away. But, if you have only requested `read:user`, then you won't have `user`.

Reply:
correct and then the only way you can read email is if you requested and received `user:email`. this branch is only active if the public email based on `read:user` (the normal request earlier) is not defined.

Reply:
Hmmmm? Are you saying only user:email is relevant and neither user or read:user makr sense?

Reply:
`read:user` is required as a basic scope to read the user profile and that is done earlier. however, this may not set email because it is not public.

to retrieve the emails whoever is running the authenticator would need to add `user:email` or `user` the broader one to scope. thus this piece of the code uses only those two to check. i'm saying `read:user` is irrelevant for this part.

Reply:
I understood `user:read` would not imply that you would be allowed to _set_ user email, but I expected it to allow you to read private user email. But, okay that isn't the case and then `user:email` is what is relevant.

This is the correct logic then